zokrates_core/src/ir/interpreter.rs

```diff
@@ -300,8 +298,28 @@ impl fmt::Display for Error {
             Error::UnsatisfiedConstraint {
                 ref left,
                 ref right,
-                ref message,
-            } => write!(f, "{}: expected {} to equal {}", message, left, right),
+                ref error,
+            } => {
+                write!(
+                    f,
+                    "{}",
+                    error
+                        .as_ref()
+                        .map(|m| m.to_string())
+                        .unwrap_or_else(|| format!(
+                            "Unsatisfied constraint: expected {} to equal {}",
```

Comment:
I can't see a case where something can fail but there is no error. But since we represent everything as `Constraint` we cannot really encode that in the Rust type system. Maybe the least bad option is to replace `unwrap_or_else` by `expect("Found an unsatisfied constraint without an attached error.")` or something